subprojects/build-cache-http/src/integTest/groovy/org/gradle/caching/http/internal/HttpBuildCacheServiceErrorHandlingIntegrationTest.groovy

```diff
@@ -58,12 +60,27 @@ class HttpBuildCacheServiceErrorHandlingIntegrationTest extends AbstractIntegrat
 
         when:
         executer.withStackTraceChecksDisabled()
-        executer.withFullDeprecationStackTraceDisabled()
+        executer.withStacktraceDisabled()
         withBuildCache().succeeds "customTask"
 
         then:
         output =~ /Could not store entry .* for task ':customTask' in remote build cache/
-        output =~ /.*org\.gradle\.caching\.BuildCacheException: Unable to store entry at .*: ${errorPattern}/
+        output =~ /Unable to store entry at .*: ${errorPattern}/
+    }
+
+    def "build cache is deactivated for the build if the connection times out"() {
+        httpBuildCacheServer.blockIncomingConnections = true
+        startServer()
+
+        when:
+        executer.withArgument("-D${SOCKET_TIMEOUT_SYSTEM_PROPERTY}=2000")
```

Comment:
Would it make sense to wait less than two seconds here?

Reply:
I set the timeout to 1 second.